terraform/context_refresh_test.go

```diff
@@ -1907,6 +1907,65 @@ data "aws_data_source" "foo" {
 	}
 }
 
+// we can't refresh managed resources that don't exist in the state
+func TestContext2Refresh_missingInstance(t *testing.T) {
+	m := testModuleInline(t, map[string]string{
+		"main.tf": `
+resource "aws_instance" "foo" {
+  count = 1
+}
+
+resource "aws_instance" "bar" {
+  count = 1
+  foo = aws_instance.foo[count.index].id
+}
+
+`,
+	})
+
+	p := testProvider("aws")
+	p.PlanResourceChangeFn = func(req providers.PlanResourceChangeRequest) (resp providers.PlanResourceChangeResponse) {
+		resp.PlannedState = req.ProposedNewState
+		return
+	}
+	p.ReadResourceFn = func(req providers.ReadResourceRequest) (resp providers.ReadResourceResponse) {
+		// resource has disappeared
+		resp.NewState = cty.NullVal(req.PriorState.Type())
+		return
+	}
+
+	state := states.NewState()
+	root := state.EnsureModule(addrs.RootModuleInstance)
```

Comment:
I think there's a better function for this right near where `EnsureModule`, to create a new state